app/lib/shared/platform.dart

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:pana/pana.dart';
+
+abstract class KnownPlatforms {
+  static const String flutter = PlatformNames.flutter;
+  static const String server = PlatformNames.server;
+  static const String web = PlatformNames.web;
+  static const List<String> everywhere = const [flutter, server, web];
+
+  static bool isPredicate(String platform) => everywhere.contains(platform);
+}
```

Comment:
We have now at least 5 different types we use for two concepts (platform identifier, platform predicate):
```
PlatformNames, KnownPlatforms, PlatformPredicate, DartPlatform, String
```
e.g. Is it obvious what the difference between `PlatformNames.everywhere` and `KnownPlatforms.everywhere` is? To me not at all (of course I can find out, but it's not obvious)!

Also there a difference between a **platform name** and a **platform** ? Both seem to be just plain `String`s.

Reply:
The platform name is simple, and it is a `String`. But their relation is complex, e.g. "runs of Flutter" can mean several things:
- requires Flutter
- runs only on Flutter
- runs on Flutter but also on other platforms.

`DartPlatform` (from pana) gives a DRY format of the above, e.g. a package either runs everywhere, or it is restricted to only a few platforms or a single one. However, the search use cases are more complex, hence the `PlatformPredicate`.

`KnownPlatforms` is a helper, so that the rest of the pub site doesn't need to depend on `pana` internal `PlatformNames` (which also has additional entries for the debug-centric `DartPlatform.description` field).